src/tools/perf/ucp_tests.cc

```diff
@@ -131,6 +131,24 @@ class ucp_perf_test_runner {
         return UCS_OK;
     }
 
+    ssize_t UCS_F_ALWAYS_INLINE wait_stream_recv(void *request)
+    {
+        size_t       length;
+        ucs_status_t status;
+
+        if (ucs_likely(!UCS_PTR_IS_PTR(request))) {
```

Comment:
why is this check needed? the function is called only if this is really a ptr. maybe should be an assertion?